src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -74,9 +83,18 @@ public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> obs
         /// Use Subscribe overload with name-based filtering if producer does NOT support context-based filtering</param>
         public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
         {
-            return isEnabled == null ?
-                SubscribeInternal(observer, null, null) :
-                SubscribeInternal(observer, name => IsEnabled(name, null, null), isEnabled);
+            IDisposable subscription;
+            if (isEnabled == null)
+            {
+                subscription = SubscribeInternal(observer, null, null);
+            }
+            else
+            {
+                Func<string, object, object, bool> localIsEnabled = isEnabled;
+                subscription = SubscribeInternal(observer, name => IsEnabled(name, null, null), localIsEnabled);
```

Comment:
There's no need to change this one from what you had in the previous commit, as `isEnabled` isn't being closed over.